### DISCUSSION TITLE
How do you deal with errors and e.g. sentry?

### DISCUSSION THREAD

**Discussion by marc-on-github** [trpc/trpc#1934] [Q&A] [answered]
I am currently using in a nextjs project the possibility to return server side errors via this way:

for example:
   throw new TRPCError({
          code: 'FORBIDDEN',
          message: 'You need min one element'
        });

so I can handle this error on the client.
We use in our project sentry to track errors. Now sentry is logging this error:
![image](https://user-images.githubusercontent.com/7252726/170210198-7dd29007-825a-4432-9a37-16ad5c9748be.png)

I wonder how you solve that. Or how do you deal with things like that?
Do you give 200 status back and a different message in the body?

Would appreciate your help very much! Thank you!

**Comment by KATT** [comment, marked as answer]
https://github.com/trpc/examples-next-prisma-starter/blob/5158b825a2c236e19ef1e7128f86021498fc833c/src/pages/api/trpc/%5Btrpc%5D.ts#L14-L22

**Reply by divmgl** [reply]
@KATT Do you have suggestions on how to do this outside of Next.js? I want to log an error into stdout in development so that I don't have to go through its details in the frontend.

**Comment by devidw** [comment]
One way I could think of would be to use a custom `errorFormatter` and as a side effect do the bug reporting.

```ts
const t = initTRPC.context<CustomContext>().create({
    errorFormatter: function({ shape, error }): any {
        Sentry.captureException(error)
        return shape
    },
})
```
